parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -401,6 +405,22 @@ public <D> FileAppender<D> build() throws IOException {
         Preconditions.checkState(fileAADPrefix == null, "AAD prefix set with null encryption key");
       }
 
+      Map<String, String> colNameToParquetPathMap =
+          type.getColumns().stream()
+              .filter(col -> col.getPrimitiveType().getId() != null)
+              .filter(
+                  col -> {
+                    int fieldId = col.getPrimitiveType().getId().intValue();
+                    return schema.findColumnName(fieldId) != null;
+                  })
+              .collect(
+                  Collectors.toMap(
+                      col -> {
+                        int fieldId = col.getPrimitiveType().getId().intValue();
+                        return schema.findColumnName(fieldId);
```

Comment:
Nit: The getId() and findColumnName() logic is used multiple times. Could we extract it into a helper method?

Reply:
Thanks @huaxingao for the review. I thought about it when I was working on a patch. The logic there is essentially findColumnName(). The getId() is to access the the field in the col. In my opinion, there is not much to extract. I can merge two filters call into one, will upload a new patch. What do you think?

Reply:
I refactored the code and uploaded a new patch, please take a look, thanks.